collector/cluster_license.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type clusterLicenseMetric struct {
+	Type   prometheus.ValueType
+	Desc   *prometheus.Desc
+	Value  func(clusterLicenseStats clusterLicenseResponse) float64
+	Labels func(clusterLicenseStats clusterLicenseResponse) []string
+}
+
+var (
+	defaultClusterLicenseLabels = []string{"cluster_license_type"}
+	defaultClusterLicenseValues = func(clusterLicense clusterLicenseResponse) []string {
+		return []string{clusterLicense.License.Type}
+	}
+)
+
+// License Information Struct
+type ClusterLicense struct {
+	logger log.Logger
+	client *http.Client
+	url    *url.URL
+
+	clusterLicenseMetrics []*clusterLicenseMetric
+}
+
+// NewClusterLicense defines ClusterLicense Prometheus metrics
+func NewClusterLicense(logger log.Logger, client *http.Client, url *url.URL) *ClusterLicense {
+	return &ClusterLicense{
+		logger: logger,
+		client: client,
+		url:    url,
+
+		clusterLicenseMetrics: []*clusterLicenseMetric{
+			{
+				Type: prometheus.GaugeValue,
+				Desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, "cluster_license", "max_nodes"),
+					"The max amount of nodes allowed by the license",
+					defaultClusterLicenseLabels, nil,
+				),
+				Value: func(clusterLicenseStats clusterLicenseResponse) float64 {
+					return float64(clusterLicenseStats.License.MaxNodes)
+				},
+				Labels: defaultClusterLicenseValues,
+			},
+			{
+				Type: prometheus.GaugeValue,
+				Desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, "cluster_license", "issue_date_in_millis"),
```

Comment:
Metrics based on times should be seconds, not millis. They should also be epoch which I believe these are. 

The new name would be "issue_date_seconds"